cfme/tests/infrastructure/test_instance_analysis.py

```diff
@@ -166,68 +170,65 @@ def _host_remove_creds():
-    base_data = provider.data.get('vm_analysis_new', {}).get('provisioning', {})
-    base_data.update(provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
-    return base_data
+
+    pdata = provider.data
+    provisioning_data = pdata.get('vm_analysis_new', {}).get('provisioning', {})
+
+    # Setup up the provisioning data for the instance/vm
+    # Will default to provisioning items under vm_analysis but if not defined
+    #   falls back to items under provider['provisioning'] key
+    if 'host' not in provisioning_data and 'host' in pdata['provisioning']:
+        provisioning_data['host'] = pdata['provisioning']['host']
+    if 'datastore' not in provisioning_data and 'datastore' in pdata['provisioning']:
+        provisioning_data['datastore'] = pdata['provisioning']['datastore']
+    if 'vlan' not in provisioning_data and 'vlan' in pdata['provisioning']:
+        provisioning_data['vlan'] = pdata['provisioning']['vlan']
+    if 'instance_type' not in provisioning_data and 'instance_type' in pdata['provisioning']:
+        provisioning_data['instance_type'] = pdata['provisioning']['instance_type']
+    if 'availability_zone' not in provisioning_data and \
+            'availability_zone' in pdata['provisioning']:
+        provisioning_data['availability_zone'] = pdata['provisioning']['availability_zone']
+    if 'security_group' not in provisioning_data and 'security_group' in pdata['provisioning']:
+        provisioning_data['security_group'] = pdata['provisioning']['security_group']
+    if 'cloud_network' not in provisioning_data and 'cloud_network' in pdata['provisioning']:
+        provisioning_data['cloud_network'] = pdata['provisioning']['cloud_network']
+
+    # If defined, tries to find cluster from provisioning, then provider definition itself
+    if provider.type == 'rhevm':
+        if 'cluster' not in provisioning_data:
+            if 'cluster' not in pdata['provisioning']:
+                provisioning_data['cluster'] = pdata['default_cluster']
+            else:
+                provisioning_data['cluster'] = pdata['provisioning']['cluster']
+
+    provisioning_data.update(
+        provider.data.get('vm_analysis_new', {}).get('vms', {}).get(analysis_type, {}))
+    return provisioning_data
 
 
 @pytest.fixture(scope="module")
 def instance(request, local_setup_provider, provider, vm_name, vm_analysis_data):
     """ Fixture to provision instance on the provider """
 
-    template = vm_analysis_data.get('image', None)
-    host_name, datastore_name = map(vm_analysis_data.get, ('host', 'datastore'))
-
-    mgmt_system = provider.get_mgmt_system()
-
-    provisioning_data = {
-        'vm_name': vm_name,
-        'host_name': {'name': [host_name]},
-        'datastore_name': {'name': [datastore_name]},
-    }
+    vm = VM.factory(vm_name, provider, template_name=vm_analysis_data['image'])
+    request.addfinalizer(lambda: cleanup_vm(vm_name, provider))
```

Comment:
might want to set this after create_on_provider, since it will fail if create_on_provider fails and vm_name was never created.